examples/realtime/nextjs-auth-presence/pages/_app.tsx

```diff
@@ -0,0 +1,39 @@
+import '../styles/globals.css'
+import { useState } from 'react';
+import type { AppProps } from 'next/app'
+import { useRouter } from 'next/router';
+import { createBrowserSupabaseClient, Session } from "@supabase/auth-helpers-nextjs";
+import { SessionContextProvider } from "@supabase/auth-helpers-react"
+
+function MyApp({ Component, pageProps }: AppProps<{
+  initialSession: Session;
+}>) {
+  const router = useRouter();
+  const [supabaseClient] = useState(() => createBrowserSupabaseClient());
+
+  supabaseClient.auth.onAuthStateChange((event, session) => {
+    switch( event) 
+    { case 'SIGNED_IN': 
+        router.push('/');
+        return;
+      case 'SIGNED_OUT':
+        router.push('/login');
+        return;
+    }
+  })
```

Comment:
This needs wrapping in a `useEffect`. For example:
```javascript
useEffect(() => {
  const {
    data: { subscription },
  } = supabaseClient.auth.onAuthStateChange((event, session) => {
    switch (event) {
      case 'SIGNED_IN':
        router.push('/');
        return;
      case 'SIGNED_OUT':
        router.push('/login');
        return;
    }
  });

  return subscription.unsubscribe;
}, []);
```

Reply:
Fixed. Wrapped onAuthStateChange inside useEffect